gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/common/impl/SimpleConfigStoreResolver.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.config.common.impl;
+
+import gobblin.config.store.api.ConfigStore;
+import gobblin.config.store.api.ConfigStoreWithResolution;
+import gobblin.config.store.api.VersionDoesNotExistException;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.UUID;
+
+import org.apache.log4j.Logger;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigValue;
+
+/**
+ * Utility class to provide {@ConfigStoreWithResolution} functionality based on given {@ConfigStore}
+ * @author mitu
+ *
+ */
+public class SimpleConfigStoreResolver implements ConfigStoreWithResolution {
+
+  private static final Logger LOG = Logger.getLogger(SimpleConfigStoreResolver.class);
+  private final ConfigStore store;
```

Comment:
I'd say, don't even implement the ConfigStoreWithResolution altogether. Stores that need the functionality can instantiate the resolver rather than extend it.